physical/postgresql.go

```diff
@@ -69,8 +69,8 @@ func newPostgreSQLBackend(conf map[string]string) (Backend, error) {
 		"put":    put_statement,
 		"get":    "SELECT value FROM " + quoted_table + " WHERE path = $1 AND key = $2",
 		"delete": "DELETE FROM " + quoted_table + " WHERE path = $1 AND key = $2",
-		"list":   "SELECT key FROM " + quoted_table + " WHERE path = $1" +
-				"UNION SELECT substr(path, length($1)+1) FROM " + quoted_table + "WHERE parent_path = $1",
+		"list": "SELECT key FROM " + quoted_table + " WHERE path = $1" +
```

Comment:
Auto-`go fmt` from my editor picked this up along the way so I added it as a distinct commit.  At one point before adding the `ServiceDiscovery` interface, I added a `Run()` method to the generic backend and touched a number of the physicals.  There's no reason this couldn't be backed out.